From c587da925b6dc45e9e464b30969ce8b95d92ca7a Mon Sep 17 00:00:00 2001
From: Fang Wang <fang.c.wang@intel.com>
Date: Tue, 7 Jul 2015 10:07:14 +0800
Subject: [PATCH 442/462] OV8858: Added 1640x1232 sensor setting

For ITS case test_yuv_jpeg_all:
Sensor need to output small size for capture.

Change-Id: I561c20dbe20e0f71d211391a0d322a1e9d273946
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-40714
Signed-off-by: Fang Wang <fang.c.wang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/63726
Approver: Yunliang Ding <yunliang.ding@intel.com>
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Approver: Jian Xu Zheng <jian.xu.zheng@intel.com>
Reviewed-by: Jian Xu Zheng <jian.xu.zheng@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: IMIN (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/i2c/ov8858.h              | 23 ++++++++++++++++++----
 1 file changed, 19 insertions(+), 4 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
index b72e7e1..3aea460 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
@@ -1239,12 +1239,27 @@ static struct ov8858_resolution ov8858_res_preview[] = {
 	},
 };
 
-/*
- * FIXME: Since ISP can't downscale image from 1640x1232 to 176x144
- * So don't add 1640x1232 setting into still mode here.
- */
 static struct ov8858_resolution ov8858_res_still[] = {
 	{
+		.desc = "ov8858_1640x1232_STILL",
+		.width = 1640,
+		.height = 1232,
+		.used = 0,
+		.regs = ov8858_1640x1232,
+		.bin_factor_x = 0,
+		.bin_factor_y = 0,
+		.skip_frames = 0,
+		.fps_options = {
+			{
+				.fps = 30,
+				.pixels_per_line = 3880,
+				.lines_per_frame = 2573,
+			},
+			{
+			}
+		},
+	},
+	{
 		.desc = "ov8858_1640x926_STILL",
 		.width = 1640,
 		.height = 926,
-- 
1.9.1

